pkg/build/gobuild.go

```diff
@@ -939,3 +998,158 @@ func (pm *platformMatcher) matches(base *v1.Platform) bool {
 
 	return false
 }
+
+type layerCache struct {
```

Comment:
non-blocking: pkg/build/gobuild.go is getting pretty long, could this go into pkg/build/cache.go or something?